src/components/CreatePostInput/index.js

```diff
@@ -14,34 +15,29 @@ import {joinTopicIntoTopicList} from '../../utils/array/ChunkArray';
 
 const CreatePostInput = ({
   allTaggedUser = [],
-  message = '',
+  message,
   topics = [],
   topicChats = [],
-  setAllTaggedUser = () => {},
-  setMessage = () => {},
-  setPositionKeyboard = () => {},
-  setTopics = () => {},
-  setTopicChats = () => {}
+  setAllTaggedUser,
+  setPositionKeyboard,
+  setTopics,
+  setTopicChats,
+  setMessage
 }) => {
   const [positionEndCursor, setPositionEndCursor] = React.useState(0);
   const [topicSearch, setTopicSearch] = React.useState([]);
   const [userTaggingSearch, setUserTaggingSearch] = React.useState([]);
   const [hashtagPosition, setHashtagPosition] = React.useState(0);
   const [positionTopicSearch, setPositionTopicSearch] = React.useState(0);
   const [lastTopicInCursor, setLastTopicInCursor] = React.useState('');
-  const [shouldUpdateHashtag, setShouldUpdateHashtag] = React.useState(0);
   const inputRef = React.useRef();
   const {formattedText, handleStateHashtag, handleStateMention, setHashtags, updateHashtag} =
     useHastagMention('');
 
   React.useEffect(() => {
-    setShouldUpdateHashtag(new Date().valueOf());
+    if (new Date().valueOf() > 0) updateHashtag(message, topics, setTopics, positionEndCursor);
   }, [message]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The default values for some props have been removed and the state variable `shouldUpdateHashtag` has been eliminated. This simplifies the code and removes unnecessary state management. However, please ensure that these changes do not break any existing functionality where these defaults or the `shouldUpdateHashtag` state were relied upon.
```diff
-  message = '',
-  setAllTaggedUser = () => {},
-  setMessage = () => {},
-  setPositionKeyboard = () => {},
-  setTopics = () => {},
-  setTopicChats = () => {}
-  const [shouldUpdateHashtag, setShouldUpdateHashtag] = React.useState(0);
+  message,
+  setAllTaggedUser,
+  setMessage,
+  setPositionKeyboard,
+  setTopics,
+  setTopicChats
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->